tests/dump_tests/module_tests/mock_sonicv2connector.py

```diff
@@ -0,0 +1,69 @@
+import json, re, os
+from utilities_common.constants import DEFAULT_NAMESPACE
+
+class MockSonicV2Connector():
+    def __init__(self, dedicated_dbs, **kwargs):
```

Comment:
Why we need to introduce a new class for mocking Sonicv2connector. Can we add new functionality to `dbconnector.py`

Reply:
Testing Use case for these modules is to read from a separate json files as opposed to other tests which read from the default config files inside mock_tables directory.    

I did see there is a dedicated_db list used inside one of the dbconnector methods 
```
global dedicated_dbs
if dedicated_dbs and dedicated_dbs.get(db_name):
        self.dbintf.redis_kwargs['db_name'] = dedicated_dbs[db_name]
else:
        self.dbintf.redis_kwargs['db_name'] = db_name
```
The only use case which uses this is db_migrator_test.py. However in that test, it was initializing the SonicV2Connector class directly inside the test, whereas in dump_module_tests it is being called in the methods defined actual classes. 

And probably because of the complex redirection used in the dbconnector.py to mock the SonicV2Connector, i was not able to get the dump_module_tests tests running on custom json files. I've also worked with stephan who introduced that change but couldn't get it running.  

Hence, decided to add a new mock class.

Reply:
I found a couple of examples of using the `dedicated_dbs` recently added recently

https://github.com/Azure/sonic-utilities/blob/master/tests/config_int_ip_test.py
https://github.com/Azure/sonic-utilities/blob/5002745beb89a99a1cdf420f34a495c2c0cb31bd/tests/mpls_test.py#L60-L67


Can you check if we can follow a similar approach here?
I feel we should try and avoid duplicate mock classes if possible

Reply:
Thanks for the info. I was able to remove the extra mock and use the default mock. I couldn't get it to directly read the custom json files, but i've handled it nevertheless.

Regarding the connection pooling optimization, i've recently observed when operating at scale, (i.e. route, had around 10k routes to parse for the dump utility) i've observed a significant amount of time spent in connect calls. With this optimization, the number of connect calls can be effectively reduced from (number_of_routes*number_of_calls_per_route) to 4 (number of db's the module is looking into) and thus offering a significant improvement in performance.
This design change infact helped me to use the default mock as well.

@SuvarnaMeenakshi  and @arlakshm , please provide your comments on the changes.